### PR TITLE
Move projects out of config.xml

### PR DESCRIPTION
**Pull request by brubo1**

### CODE REVIEW

File: modules/configuration/php/NDB_Form_configuration.class.inc [hunk 1]
```diff
@@ -135,6 +137,29 @@ class NDB_Form_Configuration extends NDB_Form
...
+        $useProjects = $config->getSetting("useProjects");
+        if ($useProjects === "true") {
+            $projectList = Utility::getProjectList();
+            $projects    = array();
+            foreach ($projectList as $ProjectID => $Name) {
+                $projects[$ProjectID] = $config->getProjectSettings($ProjectID);
+            }
+            $this->tpl_data['projects'] = $projects;
+        } else {
+            throw new Exception("Projects are not used in this study.");
```
Comment (brubo1) [inline modules/configuration/php/NDB_Form_configuration.class.inc]: ConfigurationException makes sense. Basically, the user tried to access a
page which he does not use.

File: modules/configuration/ajax/updateProject.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+* This file is used by the Configuration module to update
+* or insert values into the Config table.
```
Comment (taracampbell) [inline modules/configuration/ajax/updateProject.php]: Could you update the file comment to be a bit more specific?

File: SQL/2015-08-11-projectConfigTable.sql [hunk 3]
```diff
@@ -0,0 +1,7 @@
+
```
Comment (taracampbell) [inline SQL/2015-08-11-projectConfigTable.sql]: Just an aesthetic thing, but could you remove the blank line at the beginning of this file?

**Comment by taracampbell** [comment]
I am getting this error when I run the import script:

```PHP Notice:  Undefined offset: 1 in /Users/Tara/Sites/loris/modules/configuration/tools/import_projects.php on line 32```

File: modules/configuration/tools/import_projects.php [hunk 4]
```diff
@@ -12,32 +21,62 @@
...
+    echo ("-p -> Imports only the projects from the XML.\r\n");
+    echo ("-a -> Imports both the subprojects and the projects \r\n");
+    echo ("      from the XML.\r\n");
+    exit (2);
+}
+if ((isset($argv[1]) && $argv[1] === "-s")
+    || (isset($argv[1]) && $argv[1] === "-a")
+) {
+    foreach ($subprojs['subproject'] as $row) {
+        $windowDiff = "optimal";
```
Comment (taracampbell) [inline modules/configuration/tools/import_projects.php]: @driusan, maybe this is something for you to check.

I am still getting NULL in the WindowDifference column when it is not set in my config.xml, even though you have defined the default in this script as 'optimal'.

My subproject config.xml looks like:

```xml

```

Comment (brubo1) [inline modules/configuration/tools/import_projects.php]: PHP Notice: Undefined offset: 1 in /modules/configuration/tools/import_projects.php on line 32
Does it not just tells you that the offset is not defined.
I could do 
$option = 1;
and change the if statement at line 32 to 
if (is_null($argv[$option])

**Pull request closed without merging**